### ISSUE TITLE
Exhaustive Search Not returning results

### ISSUE THREAD

**Issue by e-lobo** [typesense/typesense#871]
## Description

Exhaustive Search is not working as expected as it used to in version 0.23

## Steps to reproduce

Notice the screenshots:

Screenshot 1
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/41537985/216006348-b3a86696-d165-4af0-ae02-1bac67719c86.png">

Screentshot 2
<img width="1058" alt="image" src="https://user-images.githubusercontent.com/41537985/216006511-7d3a8f29-920e-4da7-a8cf-ea06b9b7c7e4.png">

## Expected Behavior

we expect to see the products seen in screenshot 1 when querying the second time in screenshot 2

## Actual Behavior

we cannot see all the products which are seen in screenshot 1 in screenshot 2

## Metadata

**Typesense Version**: 0.24

**Comment by kishorenc** [comment]
@e-lobo 

Can you please email us your cluster ID to contact@typesense.org to help us figure out what's happening here?

**Comment by kishorenc** [comment]
@e-lobo 

Thanks for sharing that and for creating this issue. There has certainly been a regression here. I already have a fix for this issue on the 0.25 branch and can publish a RC build in a few days.

For now, you have to explicitly set `drop_tokens_threshold=X` where `X` is some high number to get the older behavior.

**Comment by kishorenc** [comment]
@e-lobo 

I've a fix available as a patch on 0.24 itself. Can we upgrade your cluster to this version for you to test? Do you have a staging cluster we can try first or let me know how you run Typesense locally and I can share a build (Docker or binary).

**Comment by e-lobo** [comment]
> Can we upgrade your cluster to this version for you to test?

yes please..

**Comment by jasonbosco** [comment]
@e-lobo We've upgraded your Typesense Cloud cluster. Could you give it a shot now?

**Comment by e-lobo** [comment]
looks fine although im noticing the issue below now..

searching by indexed nested fields seems off..

searching the below => `shawarma spicy` we see results which is fine i guess..

<img width="1272" alt="image" src="https://user-images.githubusercontent.com/41537985/217443941-4d474b35-3690-4ec8-9ae2-5cf2ce713626.png">

now just searching `spicy` we are not seeing some of the results from above..
<img width="1221" alt="image" src="https://user-images.githubusercontent.com/41537985/217444297-ffe2631c-14aa-4bf9-bf1a-5bd7a8a93e75.png">

please note exhaustive_search argument being sent..

**Comment by kishorenc** [comment]
@e-lobo 

By default Typesense ignores hyphens, so `Saj-Spicy` is indexed as `SajSpicy`. So that is not returned anymore for the `Spicy` query because Typesense does only prefix search and not suffix search. If you want the words to be split on hyphens and indexed as separate words then you have to configure hyphen in the collection schema as a separator this way:

```
 curl -k "http://localhost:8108/collections" -X POST -H "Content-Type: application/json" \
      -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
        "name": "coll",
        "fields": [
          {"name": "title", "type": "string" }
        ], 
        "token_separators": ["-"]
      }'
```

**Comment by e-lobo** [comment]
ah i see..
we are indexing the following symbols though as we noticed results where skew with our huge dataset. 
im not sure now.. 

<img width="414" alt="image" src="https://user-images.githubusercontent.com/41537985/217466653-5c2b184f-bcb4-4388-baaf-280ab3f72553.png">

**Comment by kishorenc** [comment]
`token_separators` is still empty. With `symbols_to_index` the hyphen will be treated as a character and be indexed but it won't solve the problem of `spicy` not matching since it will still be a suffix and not infix.

**Comment by e-lobo** [comment]
gotcha 👍

so i updated my collection as mentioned..

<img width="469" alt="image" src="https://user-images.githubusercontent.com/41537985/217476005-35026d10-7d01-44a0-96b7-27c85d1ed98f.png">

but still no results

<img width="1110" alt="image" src="https://user-images.githubusercontent.com/41537985/217476133-e7b5dd68-22a1-4cb3-ad64-73a5be66446a.png">

**Comment by kishorenc** [comment]
`symbols_to_index` also contains hyphen? That takes precedence.

**Comment by e-lobo** [comment]
i see ill need to analyze my data a bit then to see what works best..
Thanks for everything 👍

**Comment by kishorenc** [comment]
Fix is part of the just released `0.24.1`.

**Comment by ksumag** [comment]
Hi, @kishorenc. In what config file and where should I insert parameter 'exhaustive_search: true' for Exhaustive Search starts working?
